pkg/vulnerability/vulnerability.go

```diff
@@ -0,0 +1,100 @@
+package vulnerability
+
+import (
+	"bufio"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/utils"
+)
+
+const (
+	DefaultIgnoreFile = ".trivyignore"
+)
+
+type Client struct {
+	dbc db.Operations
+}
+
+func NewClient() Client {
+	return Client{
+		dbc: db.Config{},
+	}
+}
+
+func (c Client) FillAndFilter(vulns []types.DetectedVulnerability, severities []dbTypes.Severity,
+	ignoreUnfixed bool, ignoreFile string, light bool) []types.DetectedVulnerability {
+	var err error
+	var severity dbTypes.Severity
+
+	ignoredIDs := getIgnoredIDs(ignoreFile)
+	var vulnerabilities []types.DetectedVulnerability
+	for _, vuln := range vulns {
+		var vulnerability dbTypes.Vulnerability
+		if light {
+			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
+			vulnerability.Severity = severity.String()
+		} else {
+			vulnerability, err = c.dbc.GetVulnerability(vuln.VulnerabilityID)
+		}
+		if err != nil {
```

Comment:
Which error are you checking here?

Reply:
When `--light` is enabled, the error from line 41. When `--light` is disabled, the error from line 44. They are almost the same, so I made those error handling common.

```
		if light {
			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
			if err != nil {
				log.Logger.Warn(err)
				continue
			}
			vulnerability.Severity = severity.String()
		} else {
			vulnerability, err = c.dbc.GetVulnerability(vuln.VulnerabilityID)
			if err != nil {
				log.Logger.Warn(err)
				continue
			}
		}
```
We can write as the above.